data/src/main/java/com/linkedin/data/codec/AbstractJacksonDataCodec.java

```diff
@@ -213,12 +224,35 @@ public DataList readList(InputStream in) throws IOException
     return parse(_factory.createParser(in), mesg, locationMap);
   }
 
-  public void objectToJsonGenerator(Object object, JsonGenerator generator) throws IOException
+  /**
+   * Convert an object to Json format representation.
+   * @param object the object that needs to be converted
+   * @param generator the generator that could generate Json output based on the object
+   * @param mapKeyOrdered if the object contains map element (which could be nested map), can set this flag to order the
+   *                      map entries in the output according to sorted map key.
+   * @throws IOException
+   */
+  public void objectToJsonGenerator(Object object, JsonGenerator generator, boolean mapKeyOrdered) throws IOException
   {
-    Data.TraverseCallback callback = createTraverseCallback(generator);
+    Data.TraverseCallback callback = createTraverseCallback(generator, mapKeyOrdered);
     Data.traverse(object, callback);
   }
 
+  public static class JacksonTraverseCallbackMapKeyOrdered extends JacksonTraverseCallback
```

Comment:
Why do you need this class? Can't you add a flag in JacksonTraverseCallback and a second constructor to set it?

Reply:
Because I need to override the 
`orderMap` method in the new class. 

I don't think I got you. How do I set the override method in the constructor?

Reply:
after off line discussion, adopted the suggestion